tests/tb_device_mqtt_client_tests.py

```diff
@@ -82,19 +86,425 @@ def test_send_telemetry_and_attr(self):
+        }
+
+        creds = TBDeviceMqttClient.provision(
+            host="thingsboard.host",
+            provision_device_key="provision_key",
+            provision_device_secret="provision_secret",
+            hash="your_hash"
+        )
+        self.assertEqual(creds, {
+            "status": "SUCCESS",
+            "credentialsValue": "mockValue",
+            "credentialsType": "X509_CERTIFICATE"
+        })
+        mock_provision_client.assert_called_with(
+            host="thingsboard.host",
+            port=1883,
+            provision_request={
+                "provisionDeviceKey": "provision_key",
+                "provisionDeviceSecret": "provision_secret",
+                "hash": "your_hash",
+                "credentialsType": "X509_CERTIFICATE"
+            }
+        )
+
+    @patch('tb_device_mqtt.log')
+    @patch('tb_device_mqtt.sleep', autospec=True)
+    def test_subscribe_to_topic_already_connected(self, mock_sleep, mock_log):
+        self.client.is_connected = MagicMock(return_value=True)
+        self.client.stopped = False
+
+        with patch.object(self.client, '_send_request', autospec=False) as mock_send_request:
+            fake_result = MagicMock()
+            mock_send_request.return_value = fake_result
+
+            result = self.client._subscribe_to_topic("v1/devices/me/telemetry", qos=1)
+
+            mock_sleep.assert_not_called()
+            self.assertEqual(result, fake_result)
+
+            call_args, call_kwargs = mock_send_request.call_args
+            self.assertEqual(call_args[0], TBSendMethod.SUBSCRIBE)
+            self.assertIn("topic", call_args[1])
+            self.assertEqual(call_args[1]["topic"], "v1/devices/me/telemetry")
+            self.assertEqual(call_args[1]["qos"], 1)
+
+    @patch('tb_device_mqtt.log')
+    @patch('tb_device_mqtt.monotonic', autospec=True)
+    @patch('tb_device_mqtt.sleep', autospec=True)
+    def test_subscribe_to_topic_waits_for_connection_stopped(self, mock_sleep, mock_monotonic, mock_log):
+        self.client.is_connected = MagicMock()
+        self.client.stopped = False
+
+        mock_monotonic.side_effect = [0,2,5,9,12,13,14,15,16,17,18,19,20]
+
+        connect_side_effect = [False, False, False, False, False, False]
+
+        def side_effect_is_connected():
+            return connect_side_effect.pop(0) if connect_side_effect else False
+
+        self.client.is_connected.side_effect = side_effect_is_connected
+
+        def sleep_side_effect(_):
+            sleep_side_effect.counter += 1
+            if sleep_side_effect.counter == 4:
+                self.client.stopped = True
+
+        sleep_side_effect.counter = 0
+        mock_sleep.side_effect = sleep_side_effect
+
+        with patch('tb_device_mqtt.TBPublishInfo') as mock_tbpublishinfo_cls:
+            fake_info = MagicMock()
+            mock_tbpublishinfo_cls.return_value = fake_info
+
+            result = self.client._subscribe_to_topic("v1/devices/me/telemetry", qos=1)
```

Comment:
Subscribe to topic “v1/devices/me/telemetry” is not allowed from ThingsBoard side.